test/unit/org/apache/cassandra/io/sstable/metadata/MetadataSerializerTest.java

```diff
@@ -109,14 +147,60 @@ public void testHistogramSterilization() throws IOException
         }
     }
 
-    public File serialize(Map<MetadataType, MetadataComponent> metadata, MetadataSerializer serializer, Version version)
+    @Test
+    public void testSerializationWithEncryption() throws IOException
+    {
+        Map<MetadataType, MetadataComponent> originalMetadata = constructMetadata(false);
+
+        MetadataSerializer serializer = new MetadataSerializer();
+        File statsFile = serialize(originalMetadata, serializer, SSTableFormat.Type.current().info.getLatestVersion(), true);
+
+        Descriptor desc = new Descriptor(statsFile.parent(), "", "", new SequenceBasedSSTableId(0), SSTableFormat.Type.BTI);
+        try (RandomAccessReader in = RandomAccessReader.open(statsFile))
+        {
+            Map<MetadataType, MetadataComponent> deserialized = serializer.deserialize(desc, in, EnumSet.allOf(MetadataType.class));
+
+            for (MetadataType type : MetadataType.values())
+            {
+                assertEquals(originalMetadata.get(type), deserialized.get(type));
+            }
+        }
+    }
+
+    @Test
+    public void testEncryption() throws IOException
+    {
+        Map<MetadataType, MetadataComponent> originalMetadata = constructMetadata(false);
+
+        for (Version version : ImmutableList.of(BigFormat.latestVersion, TrieIndexFormat.latestVersion))
+        {
+            MetadataSerializer serializer = new MetadataSerializer();
+            File statsFile = serialize(originalMetadata, serializer, version, true);
+
+            byte[] contents = Files.toByteArray(statsFile.toJavaIOFile());
+            byte[] sought = sensitiveKey.getBytes(StandardCharsets.UTF_8);
+
+            assertEquals(version.metadataAreEncrypted(), Bytes.indexOf(contents, sought) == -1);
```

Comment:
Do we have this kind of test (that plaintext is not found in the file) for data and index files?

There's at least one problem with the current code (skipping encryption when chunk is not compressible) that would be caught by something like that.